client/my-sites/comments/comment/index.jsx

```diff
@@ -84,6 +95,33 @@ export class Comment extends Component {
 		}
 	};
 
+	getCommentOffsetTop = () => {
+		if ( ! window || `#comment-${ this.props.commentId }` !== window.location.hash ) {
```

Comment:
Can the comment list pass a more explicit prop to comment? For example `scrollIntoView: true`

Reply:
Another possibility is listening to navigation events, and setting some redux state that comment can look at